### PR TITLE
update hack/local-cluster-up.sh,make kube-scheduler use --config option

### LINKED ISSUE #58346: hack/local-up-cluster.sh passes deprecated options to scheduler
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
When I run local cluster by `hack/local-up-cluster.sh` command, I see a warning in a scheduler log:
```console
$ grep WARNING /tmp/kube-scheduler.log
W0116 17:40:03.572037    5944 server.go:158] WARNING: all flags other than --config are deprecated. Please begin using a config file ASAP.
```

**What you expected to happen**:
No such warning.

**How to reproduce it (as minimally and precisely as possible)**:
```
$ hack/local-up-cluster.sh
```

**Environment**:
```
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.0-beta1", GitCommit:"3258431", GitTreeState:"clean", BuildDate:"2018-01-08T18:34:21Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.0-alpha.1.878+a1e3b3f0533d5d-dirty", GitCommit:"a1e3b3f0533d5d211d0d8fad15005ee3cb7d67f0", GitTreeState:"dirty", BuildDate:"2018-01-16T16:22:09Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by wanghaoran1988**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #58346 

**Special notes for your reviewer**:

**Release note**:

```
None

```

### CODE REVIEW

**Comment by cblecker** [comment]
@kubernetes/sig-scheduling-pr-reviews Can someone from sig-scheduling review this?

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -823,12 +823,17 @@ EOF
       --v=${LOG_LEVEL} 2>&1 &
     PROXY_PID=$!
 
+    cat <<EOF > /tmp/kube-scheduler.yaml
```
Comment (php-coder) [inline hack/local-up-cluster.sh]: Could we remove this file after stopping? I found that `/tmp/kube-proxy.yaml` is also doesn't removed but I don't think that it's a good pattern that we should follow.

Comment (wanghaoran1988) [reply hack/local-up-cluster.sh]: @php-coder Update in the cleanup func

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -378,11 +378,15 @@ cleanup()
 
   # Check if the proxy is still running
   [[ -n "${PROXY_PID-}" ]] && PROXY_PIDS=$(pgrep -P ${PROXY_PID} ; ps -o pid= -p ${PROXY_PID})
-  [[ -n "${PROXY_PIDS-}" ]] && sudo kill ${PROXY_PIDS}
+  if [[ -n "${PROXY_PIDS-}" ]]; then
+    rm -f /tmp/kube-proxy.yaml && sudo kill ${PROXY_PIDS}
```
Comment (php-coder) [inline hack/local-up-cluster.sh]: Current code would work but it has some drawbacks:
- if `rm` returns 1, the script will never kill the processes
- if proxy server has failed to start, `PROXY_PIDS` will be empty and we leave the config file

I'd suggest to just add `rm -f /tmp/kube-proxy.yaml || :` after killing the processes.

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -432,7 +436,7 @@ function start_apiserver {
       fi
       RUNTIME_CONFIG+="scheduling.k8s.io/v1alpha1=true"
     fi
-    
+
```
Comment (php-coder) [inline hack/local-up-cluster.sh]: If it's possible, please, avoid unrelated changes.

File: hack/local-up-cluster.sh (hunk 2 above)
Comment (wanghaoran1988) [reply hack/local-up-cluster.sh]: Good catch

File: hack/local-up-cluster.sh (hunk 3 above)
Comment (wanghaoran1988) [reply hack/local-up-cluster.sh]: Just update by the hand, cannot bear the empty space in my vim setup.

File: hack/local-up-cluster.sh [hunk 4]
```diff
@@ -823,12 +825,17 @@ EOF
...
+clientConnection:
+  kubeconfig: ${CERT_DIR}/scheduler.kubeconfig
+featureGates: ${FEATURE_GATES}
+EOF
     SCHEDULER_LOG=${LOG_DIR}/kube-scheduler.log
     ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" scheduler \
       --v=${LOG_LEVEL} \
-      --kubeconfig "$CERT_DIR"/scheduler.kubeconfig \
-      --feature-gates="${FEATURE_GATES}" \
-      --master="https://${API_HOST}:${API_SECURE_PORT}" >"${SCHEDULER_LOG}" 2>&1 &
```
Comment (php-coder) [inline hack/local-up-cluster.sh]: We don't need to provide information about master anymore? I just need to ensure that it's not a bug and was removed intentionally.

Comment (wanghaoran1988) [reply hack/local-up-cluster.sh]: the master config was in the config yaml, it's removed intentionally.

**Comment by php-coder** [comment]
@wanghaoran1988 Thank you!

/lgtm

**Comment by wanghaoran1988** [comment]
@bsalamat Could you please help take a look and approve this?

File: hack/local-up-cluster.sh [hunk 5]
```diff
@@ -379,10 +379,12 @@ cleanup()
   # Check if the proxy is still running
   [[ -n "${PROXY_PID-}" ]] && PROXY_PIDS=$(pgrep -P ${PROXY_PID} ; ps -o pid= -p ${PROXY_PID})
   [[ -n "${PROXY_PIDS-}" ]] && sudo kill ${PROXY_PIDS}
+  rm -f /tmp/kube-proxy.yaml || :
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: we don't need `|| :` as this should never fail with the -f flag

File: hack/local-up-cluster.sh [hunk 6]
```diff
@@ -379,10 +379,12 @@ cleanup()
...
 
   # Check if the scheduler is still running
   [[ -n "${SCHEDULER_PID-}" ]] && SCHEDULER_PIDS=$(pgrep -P ${SCHEDULER_PID} ; ps -o pid= -p ${SCHEDULER_PID})
   [[ -n "${SCHEDULER_PIDS-}" ]] && sudo kill ${SCHEDULER_PIDS}
+  rm -f /tmp/kube-scheduler.yaml || :
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: same here

File: hack/local-up-cluster.sh [hunk 7]
```diff
@@ -823,12 +825,17 @@ EOF
       --v=${LOG_LEVEL} 2>&1 &
     PROXY_PID=$!
 
+    cat <<EOF > /tmp/kube-scheduler.yaml
+apiVersion: componentconfig/v1alpha1
+kind: KubeSchedulerConfiguration
+clientConnection:
+  kubeconfig: ${CERT_DIR}/scheduler.kubeconfig
+featureGates: ${FEATURE_GATES}
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: Has this been tested locally? It doesn't seem to match the spec for KubeSchedulerConfiguration: https://godoc.org/k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1#KubeSchedulerConfiguration

File: hack/local-up-cluster.sh (hunk 5 above)
Comment (php-coder) [reply hack/local-up-cluster.sh]: It could fail if user don't have permissions for removal.

I agree that it's a rare case but anyway, if we use `set -e` we should write all the code in a special way that is aware of for non-zero statuses.

Comment (php-coder) [reply hack/local-up-cluster.sh]: P.S. proof just in case:
```shell
$  rm -f /etc/passwd || echo rc=$?
rm: cannot remove '/etc/passwd': Permission denied
rc=1
```

Comment (cblecker) [reply hack/local-up-cluster.sh]: if we don't have write permissions to the temp dir, then this file would never get created. we're only looking to remove files that were created by this script. there are other instances of rm where we aren't doing this. I could see the argument to go through the script and change all instances at once, but I don't think we should do something different for these two instances when it's not done in the rest of the script

**Comment by wanghaoran1988** [comment]
@cblecker  Updated.

**Comment by wanghaoran1988** [comment]
/hold
Prefer to see https://github.com/kubernetes/kubernetes/pull/54211/ landed before this goes in.

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (ixdy) [reply hack/local-up-cluster.sh]: I realize this is just copying the pattern used by `/tmp/kube-proxy.yaml`, but this should probably be using `mktemp` instead of hard-coding a static filename.

**Comment by jayunit100** [comment]
Is this still something we want?  I'm all for more onfig in local-up-cluster since its still the easiest proving ground for dev/test, and probably more experimentation w/ the scheduler settings will be really good for improving docs over time.
